domain/application/state/application.go

```diff
@@ -450,23 +445,17 @@ func (st *ApplicationState) insertUnit(
 		return errors.Trace(err)
 	}
 
-	// TODO - we are mirroring what's in mongo, hence the unit name is known.
-	// In future we'll need to use a sequence to get a new unit id.
-	if args.UnitName == nil {
-		return fmt.Errorf("must pass unit name when adding a new unit for application %q", appID)
-	}
-	unitName := *args.UnitName
-	createParams.Name = unitName
+	createParams.Name = args.UnitName
 
 	if err := tx.Query(ctx, createNodeStmt, createParams).Run(); err != nil {
-		return errors.Annotatef(err, "creating net node row for unit %q", unitName)
+		return errors.Annotatef(err, "creating net node row for unit %q", args.UnitName)
 	}
 	if err := tx.Query(ctx, createUnitStmt, createParams).Run(); err != nil {
-		return errors.Annotatef(err, "creating unit row for unit %q", unitName)
+		return errors.Annotatef(err, "creating unit row for unit %q", args.UnitName)
 	}
 	if args.CloudContainer != nil {
-		if err := st.upsertUnitCloudContainer(ctx, tx, unitName, nodeUUID.String(), args.CloudContainer); err != nil {
-			return errors.Annotatef(err, "creating cloud container row for unit %q", unitName)
+		if err := st.upsertUnitCloudContainer(ctx, tx, args.UnitName, nodeUUID.String(), args.CloudContainer); err != nil {
```

Comment:
Small annoyance here. I think nodeUUID should be passed to `upsertUnitCloudContainer` as the uuid type and not a string. By the caller changing the type of the value it is starting to make decisions on behalf of the function that it is delegating decision making to.

TLDR keep the strong type all the way through until it reaches the thing that is actually responsible for translation.

Reply:
yeah, was aiming for consistency. we already have to change ID back to UUID so was expecting this would be done as part of that separate refactoring